nengo/utils/tests/test_magic.py

```diff
@@ -1,10 +1,135 @@
 import inspect
 
-from nengo.utils.magic import decorator
+from nengo.utils.magic import (
+    decorator,
+    ObjectProxyMethods,
+    ObjectProxy,
+    BoundFunctionWrapper,
+)
 
 state = None  # Used to make sure decorators are running
 
 
+def test_decorator():
+    class Test:
+        instance = "Not None"
+
+        @decorator
+        def myMethod(self):
+            pass
+
+    # TODO: debug some stuff about the decorator
+    # since lines 263-265 aren't being run
+    # wrapped = Test()
+    # wrapped.myMethod()
+    # Can't figure out how to get instance to be None
+
+
+def test_objectproxymethods():
+    """tests functions of ObjectProxyMethods"""
+    my_obj = ObjectProxyMethods()
+
+    class Test:
+        __module__ = None
+        __dict__ = None
+
+    my_obj.__wrapped__ = Test()
+
+    test = my_obj.__module__
+    assert test is None
+
+    test = my_obj.__dict__
+    assert test is None
+
+
+def test_objectproxy():
+    """tests functions of ObjectProxy"""
+
+    class Test:
+        __module__ = None
+        __dict__ = None
+        test = 1
+
+    my_proxy = ObjectProxy(Test())
+    assert (
+        str(dir(my_proxy))
+        == "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__',"
+        " '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__',"
+        " '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__',"
+        " '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__',"
+        " '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__',"
+        " '__weakref__', 'test']"
+        or str(dir(my_proxy))
+        == "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__',"
+        " '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__',"
+        " '__hash__', '__init__', '__le__', '__lt__',"
+        " '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__',"
+        " '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__',"
+        " '__weakref__', 'test']"  # Travis-ci has this one
+    )
```

Comment:
This seems really fragile. Why do we care about the exact `dir`?

Reply:
`ObjectProxy` explicitly defines the `__dir__` magic method, so to cover that we would need to call `dir`. That said, we should check the lists directly rather than comparing the string representation of those lists.